### PR TITLE
W-11680259: avoid throwing remote context error in canParse method

### PR DESCRIPTION
**Pull request by arielmirra**

### CODE REVIEW

File: shared/src/main/scala/amf/core/internal/plugins/document/graph/parser/FlattenedGraphParser.scala [hunk 1]
```diff
@@ -304,21 +304,21 @@ class FlattenedGraphParser(startingPoint: String, overrideAliases: Map[String, S
...
       )
 
     map.entries.foreach { entry =>
       val fieldUri        = expandUriFromContext(entry.key.as[String])
       val fieldValue      = entry.value
       val isAlreadyParsed = fields.exists(_.value.iri() == fieldUri)
       if (
-        !ignoredFields
-          .contains(fieldUri) && !isAlreadyParsed
+          !ignoredFields
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/core/internal/plugins/document/graph/parser/FlattenedGraphParser.scala]: why so much formatting ? shouldn't this be formatted already ?

**Comment by tomsfernandez** [comment]
The fix is great but please change how the AMFGraphPlugin is registered to only parse root files. This way we depend on declarative configuration, not imperative code logic

**Review by tomsfernandez — CHANGES_REQUESTED** [review]

**Review by nschejtman — DISMISSED** [review]

**Comment by nschejtman** [comment]
> The fix is great but please change how the AMFGraphPlugin is registered to only parse root files. This way we depend on declarative configuration, not imperative code logic

This is optional, the try-catch makes more sense as a fix

**Comment by tomsfernandez** [comment]
@arielmirra can't we change the context jsonld parser to return something a bit more meaningfull and avoid throwing the exception ?

**Comment by tomsfernandez** [comment]
@nschejtman what will happen once we implement remote context parsing ? This test will break. Exception caching is just a patch for me

**Review by tomsfernandez — APPROVED** [review]

**Pull request merged**